pkg/console/operator/status.go

```diff
@@ -0,0 +1,299 @@
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeAvailable,
+		Status:             operatorsv1.ConditionFalse,
+		Reason:             reasonNoPodsAvailable,
+		Message:            "No pods available for Console deployment.",
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionResourceSyncProgressing(operatorConfig *operatorsv1.Console, message string) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeProgressing,
+		Status:             operatorsv1.ConditionTrue,
+		Reason:             reasonSyncLoopProgressing,
+		Message:            message,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionResourceSyncNotProgressing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeProgressing,
+		Status:             operatorsv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionsManagementStateUnmanaged(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeAvailable,
+		Status:             operatorsv1.ConditionUnknown,
+		Reason:             reasonUnmanaged,
+		Message:            "The operator is in an unmanaged state, therefore its availability is unknown.",
+		LastTransitionTime: metav1.Now(),
+	})
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeProgressing,
+		Status:             operatorsv1.ConditionFalse,
+		Reason:             reasonUnmanaged,
+		Message:            "The operator is in an unmanaged state, therefore no changes are being applied.",
+		LastTransitionTime: metav1.Now(),
+	})
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeFailing,
+		Status:             operatorsv1.ConditionFalse,
+		Reason:             reasonUnmanaged,
+		Message:            "The operator is in an unmanaged state, therefore no operator actions are failing.",
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionsManagementStateRemoved(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type: operatorsv1.OperatorStatusTypeAvailable,
```

Comment:
@spadgett setting `Available:true` here, per your comment about upgrades, in reference to the discussion in `openshift/api`, which I've commented on as it may be in flux.